src/main/java/io/cryostat/agent/Registration.java

```diff
@@ -192,19 +229,25 @@ private DiscoveryNode defineSelf() throws UnknownHostException {
                         .startInstant()
                         .orElse(Instant.EPOCH)
                         .getEpochSecond();
-        URI uri;
-        if (jmxPort > 0) {
+        URI uri = callback;
+        if (preferJmx && jmxPort > 0) {
             uri =
                     URI.create(
                             String.format(
                                     "service:jmx:rmi:///jndi/rmi://%s:%d/jmxrmi",
                                     hostname, jmxPort));
-        } else {
-            uri = callback;
         }
         DiscoveryNode.Target target =
                 new DiscoveryNode.Target(
-                        realm, uri, appName, jvmId, pid, hostname, jmxPort, javaMain, startTime);
+                        realm,
+                        callback,
```

Comment:
Just want to double-check that this should be `callback` and not `uri`

Reply:
Ah, I think you're right. I'm rebuilding to test it now but this makes sense.